src/commands/Tools/topinvites.ts

```diff
@@ -0,0 +1,72 @@
+import { CommandStore, KlasaMessage, Timestamp } from 'klasa';
+import { SkyraCommand } from '../../lib/structures/SkyraCommand';
+import { Invite, MessageEmbed } from 'discord.js';
+import { UserRichDisplay } from '../../lib/structures/UserRichDisplay';
+import { getColor } from '../../lib/util/util';
+import { DeepRequired } from '../../lib/util/External/utility-types';
+import { BrandingColors } from '../../lib/util/constants';
+
+export default class extends SkyraCommand {
+
+	private inviteTimestamp = new Timestamp('MMMM d YYYY - HH:mm');
+
+	public constructor(store: CommandStore, file: string[], directory: string) {
+		super(store, file, directory, {
+			aliases: ['topinvs'],
+			cooldown: 10,
+			description: language => language.tget('COMMAND_TOPINVITES_DESCRIPTION'),
+			extendedHelp: language => language.tget('COMMAND_TOPINVITES_EXTENDED'),
+			requiredGuildPermissions: ['MANAGE_GUILD'],
+			runIn: ['text']
+		});
+	}
+
+	public async run(message: KlasaMessage) {
+		const response = await message.sendEmbed(new MessageEmbed()
+			.setDescription(message.language.tget('SYSTEM_LOADING'))
+			.setColor(BrandingColors.Secondary));
+
+		const invites = await message.guild!.fetchInvites();
+		const topTen = invites
+			.filter(inv => this.filterInvites(inv))
+			.sort((a, b) => b.uses! - a.uses!)
+			.first(10) as DeepRequired<Invite[]>;
+
+		if (topTen.length === 0) throw message.language.tget('COMMAND_TOPINVITES_NO_INVITES');
+
+		const display = this.buildDisplay(message, topTen);
+
+		await display.start(response, message.author.id);
+		return response;
+	}
+
+	private filterInvites(invite: Invite) {
```

Comment:
This method could be made into simply
```ts
return Boolean(invite.uses && invite.inviter?.username)
```

Reply:
Or, better yet
```ts
return (invite.uses && invite.inviter?.username)
```

Reply:
I dislike the latter, plus filter expects a boolean, so it'll be better performance-wise to always give it a boolean.

Also, the username check is unnecessary, user objects are never partial, this should be:

```typescript
return invite.uses > 0 && invite.inviter !== null;
```

Reply:
Couldn't that be simplified to 
```typescript
(invite.uses && invite.inviter)
```
as both `0` and `null` are falsy?

Reply:
It's unnecessary overhead and a failure here means a type mismatch, which will (and should) error afterwards rather than failing silently.

This check gives a guarantee of pure type safety.

Reply:
So... what I wrote is correct then?

Reply:
No, you wrote this:
https://github.com/kyranet/Skyra/blob/436526640f0a25cf41d7e5ed5bf0ea528b356202/src/commands/Tools/topinvites.ts#L43-L46